Classes/PKStateMachine.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PKStateMachine.swift
+//  Pods
+//
+//  Created by Gal Orlanczyk on 02/04/2017.
+//
+//
+
+import Foundation
+
+protocol IntRawRepresentable: RawRepresentable {
+    var rawValue: Int { get }
+}
+
+protocol StateProtocol: IntRawRepresentable, Hashable {}
```

Comment:
I wanted to make sure it will be understandable that this is a protocol PKState sounded more general. and we don't use 'I' like in c-sharp so protocol suffix sounded right.